lib/extensions/track.dart

```diff
@@ -68,26 +70,34 @@ extension TrackExtensions on Track {
 }
 
 extension TrackSimpleExtensions on TrackSimple {
-  Track asTrack(AlbumSimple album) {
-    Track track = Track();
-    track.name = name;
-    track.album = album;
-    track.artists = artists;
-    track.availableMarkets = availableMarkets;
-    track.discNumber = discNumber;
-    track.durationMs = durationMs;
-    track.explicit = explicit;
-    track.externalUrls = externalUrls;
-    track.href = href;
-    track.id = id;
-    track.isPlayable = isPlayable;
-    track.linkedFrom = linkedFrom;
-    track.name = name;
-    track.previewUrl = previewUrl;
-    track.trackNumber = trackNumber;
-    track.type = type;
-    track.uri = uri;
-    return track;
+  Future<Track> asTrack(AlbumSimple album, ref) async {
+    try {
+      final spotify = ref.read(spotifyProvider);
+      return await spotify.invoke((api) => api.tracks.get(id!));
```

Comment:
This will make a request for each track of an album. Instead, just delete this extension method and replace it with [get-several-track](https://developer.spotify.com/documentation/web-api/reference/get-several-tracks) (`spotify.tracks.list(<ids>)` for dart)

Or create an extension method on `Iterable<TrackSimple>` where you map the ids of the `TrackSimple` then get the tracks using  `spotify.tracks.list(<ids>)`

`TrackSimple.asTrack` is used in 
- [lib/modules/album/album_card.dart](https://github.com/KRTirtho/spotube/blob/0ec9f3535bf9c927d81cb67e62aa35da36648c98/lib/modules/album/album_card.dart#L57)
- [lib/provider/spotify/album/tracks.dart](https://github.com/KRTirtho/spotube/blob/0ec9f3535bf9c927d81cb67e62aa35da36648c98/lib/provider/spotify/album/tracks.dart#L36)